### PR TITLE
Refactor url [EPIC]

### PR DESCRIPTION
**Pull request by Alastair-L**
# Issue #:
https://github.com/beyondessential/tupaia-backlog/issues/770

### Changes:
Improves the way the url is specified.

### There are a series of PRs planned for this:

- [x] [Refactor Selectors](https://github.com/beyondessential/tupaia/pull/1050)
- [x] [Create Base Structure](https://github.com/beyondessential/tupaia/pull/1053)

Use url to store:
- [x] [Navigation](https://github.com/beyondessential/tupaia/pull/1148) (project and org unit)
- [x] [Dashboard](https://github.com/beyondessential/tupaia/pull/1194) (should be small)
- [x] [Map overlay](https://github.com/beyondessential/tupaia/pull/1181)
- [x] [Enlarged Dialog](https://github.com/beyondessential/tupaia/pull/1210)
- [x] [Exports](https://github.com/beyondessential/tupaia/pull/1238)
- [x] [UserPage ](https://github.com/beyondessential/tupaia/pull/1216)(password reset and verify email)
-  ~Project page~ Own card: https://github.com/beyondessential/tupaia-backlog/issues/604
- [x] [Ensure back and forward buttons work ](https://github.com/beyondessential/tupaia/pull/1244)
- [x] [Ensure GO_HOME works](https://github.com/beyondessential/tupaia/pull/1246)
- [x] ~Ensure the mobile site still works~ <- I have done a smoke test, and it seems fine!

### CODE REVIEW

**Comment by Alastair-L** [comment]
Almost done! A few points left to address please @kael89 🙏 
## Commits not yet reviewed:
There is still a bit of unreviewed code from me keeping this branch up to date with dev (below)

### Messy dev merges:
https://github.com/beyondessential/tupaia/pull/1063/commits/276dd9e4ef9e46db663457afdcc9a764d002e327
https://github.com/beyondessential/tupaia/pull/1063/commits/b989d163eaf30492b58c60541a4b6ba9bcb7c5b7
https://github.com/beyondessential/tupaia/pull/1063/commits/4bf261b0b450cd6253237590d33bfef8d4e5eb77
https://github.com/beyondessential/tupaia/pull/1063/commits/b9793da5abbd7f10bbb229a150cf9cd0f1f6e2b5
https://github.com/beyondessential/tupaia/pull/1063/commits/4a97aa22a29dbed8183513635545401031853f82
### Merge conflict fixes:
https://github.com/beyondessential/tupaia/pull/1063/commits/55be3fd7f935d97aeebae43bcf10034f413b87a6
https://github.com/beyondessential/tupaia/pull/1063/commits/9b0d4a0cdc417f5502791fd42f9771fadbbdf715
https://github.com/beyondessential/tupaia/pull/1063/commits/263859af2b2b3e297a460c2f0677ef59163c5980
https://github.com/beyondessential/tupaia/pull/1063/commits/48ae1bf2feaceca3bf438baa9c0c65d4fda2951b
https://github.com/beyondessential/tupaia/pull/1063/commits/e581abc678b5ff7bc2adf4fe8c5632432a333e62
https://github.com/beyondessential/tupaia/pull/1063/commits/7b7ebed67ca34a2da0e22723eb03d12a3f9709ae
### Other:
https://github.com/beyondessential/tupaia/pull/1063/commits/1f682c0620ad635756ace1327d0498a6c9d1fec3
### Of those, please review at least:
https://github.com/beyondessential/tupaia/pull/1063/commits/1f682c0620ad635756ace1327d0498a6c9d1fec3
https://github.com/beyondessential/tupaia/commit/e581abc678b5ff7bc2adf4fe8c5632432a333e62

The others, I believe, are simple enough to not need a review, but that's up to you.
## Unit tests
In terms of tests, should I skip all the failing ones, or leave them as is? 
Test summary on `dev`:
image
Test summary on `url-refactor`:
image

Both failing tests and all 3 skipped tests on the feature branch are failing on dev. Shall we just leave 5 failing tests in and call it an improvement, or is there a better way?
## Manual testing/what's next?
Once the above 2 points are addressed, can I mark this ready for test, and write up a list of changes to functionality? I believe that we shouldn't limit the scope of testing beyond the 'frontend' as all parts of the site were touched in at least some small way, e.g.:
- login
- disaster
- navigation
- map
- search
- etc.
Do you agree?

File: packages/web-frontend/src/selectors/measureSelectors.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      ...data,
+      ...getMeasureDisplayInfo({ ...data }, options, hiddenMeasures),
+    };
+  },
+)((_, __, ___, organisationUnitCode) => organisationUnitCode);
+
+const getOrgUnitFromMeasureData = (measureData, code) =>
+  measureData.find(val => val.organisationUnitCode === code);
+
+// TODO: Refactor to be more general
```
Comment (kael89) [inline packages/web-frontend/src/selectors/measureSelectors.js:48]: Should we address this item? No need to, just checking

**Comment by kael89** [comment]
Almost there @Alastair-L !! That's exciting. About your points:

**Commits not yet reviewed:** Had a look, they look fine (just left a comment/question)
**Unit tests:** Let's add a `.skip` to the ones that currently fail in `dev`. Could you please create a card for fixing those tests? And I'll try to prioritise it (outside the url-refactor work)
**Manual testing/what's next:** I agree, feel free to take some time to write useful notes to the testers. We should indeed not limit test to the frontend, but let's add notes on what is new functionality and what is regression testing

File: packages/web-frontend/src/selectors/measureSelectors.js (hunk 1 above)
Comment (Alastair-L) [reply packages/web-frontend/src/selectors/measureSelectors.js]: I have removed the TODO. I don't really think it is necessary.

**Review by kael89 — APPROVED** [review]
🎉 💯

**Pull request merged**